### ISSUE TITLE
Sort order of related results

### ISSUE THREAD

**Issue by edgartdata** [ycba-cia/blacklight-collections2#216]
For "The Poems of Thomas Gray", each side of the page is numbered (recto gets 1, and verso 2) and the set of results does not respect that numbering:
https://ycba-collections-dev.herokuapp.com/?utf8=%E2%9C%93&search_field=all_fields&q=%22the+poems+of+thomas+gray%22

The first image that appears in this set of results is not the first page in the book.

**Comment by edgartdata** [comment]
@yulgit1  another example of that is the prints by Lynette Yiadom-Boakye: 
https://collections.britishart.yale.edu/?f%5Bcollection_ss%5D%5B%5D=Prints+and+Drawings&q=lynette&search_field=all_fields 

![Related objects in wrong sort order](https://user-images.githubusercontent.com/4093644/88560315-f7d69e80-cffb-11ea-973f-72df002e41c2.png)

**Comment by yulgit1** [comment]
Maybe manually create a list that adds a sortnumber field to the index.  99% of the objects would just have a sortnumber=1.  For this Boakye and Grey poems, map the TMS ID to the sortnumber.

So for Boakye (I=tms:73484=1, II=tms:74614=2,III=tms:74615=3, etc.)

Something for a student maybe, to create a mapping file (1st column tmsid,2nd column sort order) that could be loaded into a database that gets queried indexing time, giving a sortnumber upon match, or 1 if tmsid not in the database (most objects).

**Comment by edgartdata** [comment]
@yulgit1 so create this list outside of TMS? 

I can see how this page/item sequencing information could also be helpful for IIIF, and ideally both BL and IIIF would use the sequencing info from the same source, so that we don't have to maintain it in multiple systems ((BL index and TMAN is it for IIIF?). If we decide to also use it in IIIF, I wonder what system would be the best source for the sequencing info? TMS? TMAN? NetX? something else?

**Comment by yulgit1** [comment]
TMAN has a rank field within an object that I believe IIIF manifests use for ordering.  But these examples here are individual objects each with their own manifest (well the Boakye isn't Iiif at all due to CDS restrictions), so this would be more like a manifest collection object.  

We could create the sorting list outside of TMS, but ideally it would be better if a field in TMS->LIDO could be leveraged, like a serial order field.

**Comment by edgartdata** [comment]
@edgartdata Select a TMS numeric field for integers and develop LIDO structure.

**Comment by edgartdata** [comment]
@azaroth42 Now that you have a good sense of our LIDO records, what LIDO element would you recommend we use for this? I can imaging that we would also want to vend this page sequencing data to our IIIF manifests as well, which would help with ordering the many images of a sketchbook in the viewer (@yulgit1 can you remind us of how the image order is currently decided in our manifests?)
As a heads up there is the use case of scholars wanting to re-order the sequence of pages of sketchbooks when they know the original order may have been tampered with.

**Comment by yulgit1** [comment]
@edgartdata FIFO from CDS

https://deliver.odai.yale.edu/info/repository/YCBA/object/34/type/2

CDS uses TMAN cdsprimary and cdsrank

But LIDO doesn't vend info on individual media (images) for an object, it just shows CDS and manifest.

**Comment by edgartdata** [comment]
@yulgit1 You are right: what I am after is a way to have individual TMS records sorted in a specific way in BL, such as the series of prints by Lynette Yiadom-Boakye above. Sorry I mudded the waters here.

**Comment by edgartdata** [comment]
We will use a TMS flex field to capture the numeric display order of the object in the group/series AND the context in which it applies, i.e. group/series.
The rationale for this is to ensure that the numeric display order is not used in any other contexts.
@edgartdata will create the flex field in TMS, expand our use of the LIDO structure to vend it to BL.
Testing will happen week November 2...depending on the news with the TMS&CS upgrades.
